at_secondary/at_persistence_secondary_server/lib/src/notification/at_notification_keystore.dart

```diff
@@ -87,14 +90,51 @@ class AtNotificationKeystore
 
   @override
   Future<bool> deleteExpiredKeys() async {
-    // TODO: implement deleteExpiredKeys
-    return Future.value(false);
+    var result = true;
+    try {
+      var expiredKeys = await getExpiredKeys();
+      if (expiredKeys.isNotEmpty) {
+        expiredKeys.forEach((element) {
+          remove(element);
+        });
+        result = true;
+      }
+    } on Exception catch (e) {
+      result = false;
+      _logger.severe('Exception in deleteExpired keys: ${e.toString()}');
+      throw DataStoreException(
+          'exception in deleteExpiredKeys: ${e.toString()}');
+    } on HiveError catch (error) {
+      _logger.severe('HiveKeystore get error: $error');
+      throw DataStoreException(error.message);
+    }
+    return result;
   }
 
   @override
   Future<List<dynamic>> getExpiredKeys() async {
-    // TODO: implement getExpiredKeys
-    return <dynamic>[];
+    var expiredKeys = <String>[];
+    try {
+      var now = DateTime.now().toUtc();
+      var keys = _getBox().keys;
+      var expired = [];
+      await Future.forEach(keys, (key) async {
+        var value = await get(key);
+        if (value != null &&
+            value.expiresAt != null &&
+            value.expiresAt!.isBefore(now)) {
+          expired.add(key);
+        }
+      });
+      expired.forEach((key) => expiredKeys.add(Utf7.encode(key)));
+    } on Exception catch (e) {
+      _logger.severe('exception in hive get expired keys:${e.toString()}');
+      throw DataStoreException('exception in getExpiredKeys: ${e.toString()}');
+    } on HiveError catch (error) {
```

Comment:
Do we have to reverse the order of exceptions

Reply:
no..error and exception are part of different hierarchy in dart